openeogeotrellis/backend.py

```diff
@@ -1153,6 +1156,98 @@ def request_costs(
 
 
 class GpsProcessing(ConcreteProcessing):
+
+    @staticmethod
+    def derive_temporal_extent(collection_id: str, constraints: dict, env: EvalEnv) -> List[Optional[str]]:
+        catalog = env.backend_implementation.catalog
+        metadata_json = catalog.get_collection_metadata(collection_id=collection_id)
+        metadata = GeopysparkCubeMetadata(metadata_json)
+
+        temporal_extent_constraints = constraints.get("temporal_extent")
+
+        catalog_temporal_extent = metadata.get("extent", "temporal", "interval", default=None)
+        outer_bounds = [None, None]
+        if catalog_temporal_extent:
+            for extent in catalog_temporal_extent:
+                if extent[0]:
+                    if outer_bounds[0] is None:
+                        outer_bounds[0] = extent[0]
+                    else:
+                        outer_bounds[0] = min(outer_bounds[0], extent[0])
+                if extent[1]:
+                    if outer_bounds[1] is None:
+                        outer_bounds[1] = extent[1]
+                    else:
+                        outer_bounds[1] = max(outer_bounds[1], extent[1])
+        if temporal_extent_constraints is None:
+            temporal_extent = outer_bounds
+        else:
+            # take the intersection of outer_bounds and temporal_extent
+            beginnings = []
+            if outer_bounds[0]:
+                beginnings.append(outer_bounds[0])
+            if temporal_extent_constraints[0]:
+                beginnings.append(temporal_extent_constraints[0])
+            if not beginnings:
+                beginnings.append(None)
+
+            ends = []
+            if outer_bounds[1]:
+                ends.append(outer_bounds[1])
+            if temporal_extent_constraints[1]:
+                ends.append(temporal_extent_constraints[1])
+            if not ends:
+                ends.append(None)
+
+            temporal_extent = (
+                max(beginnings),  # ISO date is sortable like a string
+                min(ends),
+            )
+        return temporal_extent
+
+    @staticmethod
+    def estimate_number_of_temporal_observations(collection_id: str,
+                                                 constraints: dict,
```

Comment:
Same question if this conceptually belongs under GpsProcessing

Reply:
Also moved to GeoPySparkLayerCatalog